src/NServiceBus.AcceptanceTesting/Support/PublisherMetadata.cs

```diff
@@ -10,32 +11,46 @@ public class PublisherMetadata
     {
         public IEnumerable<PublisherDetails> Publishers => publisherDetails.Values;
 
-        public void RegisterPublisherFor<T>(Type endpoint)
+        public void RegisterPublisherFor<T>(string endpointName)
         {
             PublisherDetails publisher;
+            if (!publisherDetails.TryGetValue(endpointName, out publisher))
+            {
+                publisher = new PublisherDetails(endpointName);
+
+                publisherDetails[endpointName] = publisher;
+            }
 
-            if (!publisherDetails.TryGetValue(endpoint, out publisher))
+            publisher.RegisterOwnedEvent<T>();
+        }
+
+        public void RegisterPublisherFor<T>(Type endpointType)
+        {
+            var endpointName = Conventions.EndpointNamingConvention(endpointType);
+
+            PublisherDetails publisher;
```

Comment:
Couldn't this just call the string overload here instead of duplicating the code?